sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -1092,34 +1093,51 @@ private static long toEpochMicros(Instant timestamp) {
   }
 
   @VisibleForTesting
-  public static TableRow tableRowFromMessage(Message message, boolean includeCdcColumns) {
+  public static TableRow tableRowFromMessage(
+      Message message, boolean includeCdcColumns, Predicate<String> includeField) {
+    return tableRowFromMessage(message, includeCdcColumns, includeField, "");
+  }
+
+  public static TableRow tableRowFromMessage(
+      Message message,
+      boolean includeCdcColumns,
+      Predicate<String> includeField,
+      String namePrefix) {
```

Comment:
What is this prefix actually used for? and is it relevant to this PR?

Reply:
The fieldname passed into the predicate. This allows us to handle nested fields, and pass in a fully-qualified field name (e.g. top.middle.leaf)

Reply:
Makes sense. Can we add a test case for nested fields? Also would be helpful to include this behavior in the description for `withPropagateSuccessfulStorageApiWrites()`

Reply:
Test added and description added to javadoc